src/common/registry/fetch/key-batch.fetch.ts

```diff
@@ -8,12 +8,16 @@ import { KEYS_LENGTH, SIGNATURE_LENGTH } from './key-batch.constants';
 @Injectable()
 export class RegistryKeyBatchFetchService {
   constructor(
-    @Inject(REGISTRY_CONTRACT_TOKEN)
-    private contract: Registry,
-
-    private operatorsService: RegistryOperatorFetchService,
+    protected readonly operatorsService: RegistryOperatorFetchService,
+    @Inject(REGISTRY_CONTRACT_TOKEN) private contract: Registry,
   ) {}
 
+  private getContract(moduleAddress: string) {
+    return this.contract.attach(moduleAddress);
```

Comment:
Good choice!
It works like this:
```
attach(addressOrName: string): Contract {
        return new (<{ new(...args: any[]): Contract }>(this.constructor))(addressOrName, this.interface, this.signer || this.provider);
    }
```
Let's remove this line:
`return Registry__factory.connect(moduleAddress, this.provider);`